### DISCUSSION TITLE
Support for ProductCollectionSortKeys and ProductImageSortKeys

### DISCUSSION THREAD

**Discussion by jc745** [gatsbyjs/gatsby#32097] [Ideas / Feature Requests]
Adding support for [ProductCollectionSortKeys](https://shopify.dev/docs/admin-api/graphql/reference/products-and-collections/productcollectionsortkeys) and [ProductImageSortKeys](https://shopify.dev/docs/admin-api/graphql/reference/products-and-collections/productimagesortkeys) would be welcome additions.

**ProductCollectionSortKeys**

You can imagine a situation where filtering collections of products on the frontend becomes much more feasible using `ProductCollectionSortKeys` (mentioned in gatsbyjs/gatsby-source-shopify#112).

 **ProductImageSortKeys**
 
In (limited) testing it seems that Shopify's Admin API returns product images out of their natural order, whereas the Storefront API returns them in their natural order, based on how the product's images are positioned in the Shopify Admin UI.

Implementing `ProductImageSortKeys` would allow for proper ordering when using the Admin API, though I think defaulting to the `POSITION` value would be sufficient? 

Example queries:

```
ProductCollectionSortKeys

query {
  collectionByHandle(handle: "classic"){
    products(first: 10, sortKey: BEST_SELLING) {
      ...
    }
  }
}
```

```
ProductImageSortKeys

query {
  productByHandle(handle: "snare-boot") {
    images(sortKey: POSITION) {
      ...
    }
  }
}

```

**Comment by zamson** [comment]
+1 for this, @ jc745 did manage a way to sort the images in a way that reflects Shopify admin?

**Comment by bashirdk** [comment]
+1 
I have images coming in the wrong order in my production site. Would love for a way to get the pictures in the right order without having to do some janky code hack in the UI (which wouldn't be ideal as we add more products)

**Reply by albatrocity** [reply]
This would be great at the product variant level as well. My use case requires querying a collection to show its products. I also have to get the variants for each product in order to facilitate a quick-shop sort of experience. Unfortunately I can't force a sort on the variants so their order changes per build, essentially breaking incremental builds.

(sorry to reply here, didn't feel like submitting another "answer" was appropriate)

**Comment by jackonawalk** [comment]
Hi there, we're currently testing the next major version of gatsby-source-shopify. This may address your issue. Learn more about upgrading: https://www.notion.so/gatsbyjs/Instructions-for-upgrading-to-Shopify-7-e5cb1068b35b4de7ba7e9ffb160aad26

**Comment by DanielRiveraHQ** [comment]
Do you have any suggestions on how to solve this? Upgrading to 7 or 8 still doesn't include a way to order the images.
